src/lib/user/useIsOperationAllowed.spec.ts

```diff
@@ -0,0 +1,92 @@
+            authorities: ['F_DATAELEMENT_PUBLIC_ADD'],
+        },
+        {
+            type: SchemaAuthorityType.CREATE_PRIVATE,
+            authorities: ['F_DATAELEMENT_PRIVATE_ADD'],
+        },
+        {
+            type: SchemaAuthorityType.DELETE,
+            authorities: ['F_DATAELEMENT_DELETE'],
+        },
+    ]
+    const userAuthorities = new Set(['F_DATAELEMENT_PUBLIC_ADD'])
+    it('should return true if the user has the required authorities', () => {
+        const allowed = isOperationAllowed(
+            SchemaAuthorityType.CREATE_PUBLIC,
+            dataElementSchemaAuthorities,
+            userAuthorities
+        )
+
+        expect(allowed).toBe(true)
+    })
+
+    it('should return true if user has the ALL authority', () => {
+        const userAuthorities = new Set(['ALL'])
+        const allowed = isOperationAllowed(
+            SchemaAuthorityType.CREATE,
+            dataElementSchemaAuthorities,
+            userAuthorities
+        )
+        expect(allowed).toBe(true)
+    })
+
+    it('should return true if operation is CREATE, and the schema has CREATE_PUBLIC', () => {
+        const userAuthorities = new Set(['F_DATAELEMENT_PUBLIC_ADD'])
+        const allowed = isOperationAllowed(
+            SchemaAuthorityType.CREATE,
+            dataElementSchemaAuthorities,
+            userAuthorities
+        )
+        expect(allowed).toBe(true)
+    })
+
+    it('should return true if operation is CREATE, and the schema has CREATE_PRIVATE', () => {
+        const userAuthorities = new Set(['F_DATAELEMENT_PRIVATE_ADD'])
+        const dataElementSchemaAuthorities: SchemaAuthorities = [
+            {
+                type: SchemaAuthorityType.CREATE_PRIVATE,
+                authorities: ['F_DATAELEMENT_PRIVATE_ADD'],
+            },
+        ]
+        const allowed = isOperationAllowed(
+            SchemaAuthorityType.CREATE,
+            dataElementSchemaAuthorities,
+            userAuthorities
+        )
+        expect(allowed).toBe(true)
+    })
+
+    it('should return false if the user does not have the required authorities', () => {
+        const userAuthorities = new Set(['F_PROGRAM_PRIVATE_ADD'])
+        const allowed = isOperationAllowed(
+            SchemaAuthorityType.CREATE,
+            dataElementSchemaAuthorities,
+            userAuthorities
+        )
+        expect(allowed).toBe(false)
+    })
+
+    it('should return false if schema does not have given operation', () => {
+        const userAuthorities = new Set(['F_DATAELEMENT_PRIVATE_ADD'])
+        const dataElementSchemaAuthorities: SchemaAuthorities = [
+            {
+                type: SchemaAuthorityType.CREATE_PRIVATE,
+                authorities: ['F_DATAELEMENT_PRIVATE_ADD'],
+            },
+        ]
+        const allowed = isOperationAllowed(
+            SchemaAuthorityType.DELETE,
+            dataElementSchemaAuthorities,
+            userAuthorities
+        )
+        expect(allowed).toBe(false)
+    })
```

Comment:
I've added a few test cases for empty sets. However, typescript would prevent calling this with `null`. Of course it could be `null` since it's data from the API, but in that case we have bigger problems.